### PR TITLE
Allow configuration of internal parameters

### PR DESCRIPTION
**Pull request by falvaradorodriguez**
Closes https://github.com/safe-global/safe-eth-py/issues/784

**To be considered:**

-  Some of this task was done in these previous PRs:
     https://github.com/safe-global/safe-eth-py/pull/209
     https://github.com/safe-global/safe-eth-py/pull/641

- I have checked some timeouts that were hardcoded, I would like to check with the team what other settings are missing.

### CODE REVIEW

File: gnosis/safe/api/base_api.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ def __init__(
         network: EthereumNetwork,
         ethereum_client: Optional[EthereumClient] = None,
         base_url: Optional[str] = None,
-        request_timeout: int = 10,
+        request_timeout: int = int(os.environ.get("SAFE_BASE_API_REQUEST_TIMEOUT", 10)),
```
Comment (moisses89) [inline gnosis/safe/api/base_api.py:28]: I think that would be better define this timeout in the childs, intead here. 
I mean something like `SAFE_TRANSACTION_SERVICE_REQUEST_TIMEOUT`, and the same for the relay, but not sure if will be continued.

Comment (Uxio0) [reply gnosis/safe/api/base_api.py:28]: Relay is discontinued, we should remove it. But I agree on defining this variable for every child

Comment (falvaradorodriguez) [reply gnosis/safe/api/base_api.py]: Added in TransactionServiceApi. RelayServiceApi will be removed in this [PR](https://github.com/safe-global/safe-eth-py/pull/949).

**Review by Uxio0 — APPROVED** [review]

**Comment by Uxio0** [comment]
@falvaradorodriguez Could you add documentation for this when it's merged?

**Comment by falvaradorodriguez** [comment]
> @falvaradorodriguez Could you add documentation for this when it's merged?

@Uxio0 updated [here](https://github.com/safe-global/safe-eth-py/pull/947/commits/8c06d42686394b8b201df34038594553b67dbf5e)

**Pull request merged**